dev/MAINTAINING.md

```diff
@@ -26,71 +26,98 @@
     * Tutorial has been updated for new changes
     * Documents in the `docs` directory are consistent with new changes
     * All combinators have time and space complexity annotations
     * All combinators have `since` annotation
+    * Modules that were renamed or APIs that moved to other modules should be
+      marked as "since" the current release.
+    * Check all released modules to ensure that they do not have any leftover
+      `Internal` or `Pre-release` annotations.
     * Unreleased combinators should be marked with `Pre-release` or
       `Internal` annotations
+    * Streamly homepage is updated with the release docs
+      * FileSystem.Event.Windows/Darwin modules require special handling
 
 * _Build and Test_:
 
     * All CIs are green
+    * Run tests using bin/test.sh to test with memory restrictions
     * Manually build and test for all flags (esp. `dev` flag) not covered by CIs
-    * Test latest GHC version with -O0 and -O1
+    * Test prime GHC version with -O0 and -O1
 
 * _Benchmarks_:
 
     * Check regressions from previous release
-    * Run benchmarks with large stream size (`bench.sh -- long`) to
+    * Run benchmarks with large stream size (`bench.sh --long`) to
       check for space leaks and to ensure constant memory usage for streaming
       operations.
     * Run benchmarks with `dev` flag on. Some `fileio` benchmarks are disabled
       in regular runs.
     * Check comparative benchmarks using streaming-benchmarks
 
-* _Examples_:
+* _Dependencies_:
+
+    * Make sure all dependency bounds can use latest versions
+    * Update `stack.yaml` to latest stable resolver, cleanup extra-deps
+    * Make sure fusion-plugin is up to date and uploaded
+    * Check the dependency footprint of the library. See
+      [Generating Dependency Graphs](##generating-dependency-graphs) section.
+
+* _Dependent Packages_:
 
     * Update
       [streamly-examples](https://github.com/composewell/streamly-examples)
       to make sure it runs with the latest release.
     * Check the performance of examples where applicable
+    * Update streamly-bytestring
+    * Consider updating other packages e.g. streamly-process, streamly-lz4
 
 * _Update Package Metadata:_
 
-    * Update `stack.yaml` to latest stable resolver, cleanup extra-deps
     * Make sure the description in cabal file is in sync with README and other docs
     * Make sure CI configs include last three major releases of GHC in CI testing.
-    * Update `tested-with` field
-    * Make sure all dependency bounds can use latest versions
+    * Update GHC `tested-with` field
+    * Update `docs/building.md` with the distributions tested with
     * Make sure any additional files are added to `extra-source-files` in cabal
-      file
+      file. Artifacts required for build, test, benchmarks, docs, licenses
+      should be packaged. Build environment customization may or may not be
+      packaged.
```

Comment:
docs should ideally be added to `extra-doc-files`. They are packaged along with haddocks.